### PR TITLE
feat(cli): midealocal CLI tool

### PR DESCRIPTION
**Pull request by rokam**
## Summary by CodeRabbit

- **New Features**
  - Introduced a CLI tool for interacting with Midea local devices.
  - Added device discovery, message loading, and protocol downloading functionalities to the CLI tool.
  - Configured a Python debugger for easier module discovery and debugging.
  - Added constants for application account and password configuration.
  - Added a console script entry point for the CLI tool.

- **Chores**
  - Added `midea-local.json` to `.gitignore`.
  - Updated `requirements.txt` with the `platformdirs` package.

### CODE REVIEW

**Comment by chemelli74** [comment]
Why not update/enrich library_test.py that has the same scope ?

**Comment by rokam** [comment]
I'd like to deprecate library_test and provide a tool to better understand and debug the library.

**Comment by chemelli74** [comment]
> The library_test provides no real-case use and is just a list of functions to be called.

Actually that's not true.
I use it with users to test login, keys, local connection.
Which are the main pain currently.

For sure is not feature complete and needs more love to test all functions, but so far is already useful.

What we should do is improve the token/key management in the library.

**Comment by rokam** [comment]
So, let's move the functionalities to CLI. After installing the midea-local lib, the user could call midealocal <command> <parameter>.

The library_test does an overall test, unable to provide options for the user.

**Comment by wuwentao** [comment]
thanks for add this tools, this should be a useful feature in lib side.
also expect it can decode the raw hex msg, read and decode via source code is really not a good solution,  we should check the args one by one ans need more debug effect and time cost.
but it's not a urgent requirement, we can do it one by one in future.

and also add detail comments for future changes and quickly understand the process with different developer.

File: midealocal/cli.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+
+def _message(args: Namespace) -> None:
+    """Load message into device."""
+    device_type = int(args.message[2])
+
+    device = device_selector(
+        device_id=0,
+        name="",
+        device_type=device_type,
+        ip_address="192.168.192.168",
```
[the remark this answers is not in the document]
Comment (rokam) [reply midealocal/cli.py:81]: Yep, IP address isn't used to decode the message, as it is part of input params.

[the remark this answers is not in the document]
Comment (chemelli74) [reply midealocal/cli.py:81]: so if it's part of input param, let's use what the user specify ;-)

[the remark this answers is not in the document]
Comment (rokam) [reply midealocal/cli.py:98]: It isn't part of input param. it just create the object to see if the information is stored inside the device. No communication is made.

**Review by chemelli74 — APPROVED** [review]

**Comment by chemelli74** [comment]
Maybe remove library_test in this PR as you want to deprecate it

**Pull request merged**